src/Processors/QueryPipeline.h

```diff
@@ -87,6 +87,12 @@ class QueryPipeline
             size_t max_threads_limit = 0,
             Processors * collected_processors = nullptr);
 
+    /// Add other pipeline and execute it before current one.
+    /// Pipeline must have same header.
+    void addDelayingPipeline(QueryPipeline pipeline);
```

Comment:
It's not clear which of them is delayed.  We could make a better name. Something like `addPipelineBefore()` or `addPipelineAfter()`.